source/extensions/tracers/opentelemetry/resource_detectors/resource_provider.cc

```diff
@@ -16,10 +17,21 @@ bool isEmptyResource(const Resource& resource) { return resource.attributes_.emp
 Resource createInitialResource(const std::string& service_name) {
   Resource resource{};
 
-  // Creates initial resource with the static service.name attribute.
+  // Creates initial resource with the static service.name and telemetry.sdk.* attributes.
   resource.attributes_[std::string(kServiceNameKey.data(), kServiceNameKey.size())] =
       service_name.empty() ? std::string{kDefaultServiceName} : service_name;
 
+  resource
+      .attributes_[std::string(kTelemetrySdkLanguageKey.data(), kTelemetrySdkLanguageKey.size())] =
+      kDefaultTelemetrySdkLanguage;
+
+  resource.attributes_[std::string(kTelemetrySdkNameKey.data(), kTelemetrySdkNameKey.size())] =
+      kDefaultTelemetrySdkName;
+
+  resource
+      .attributes_[std::string(kTelemetrySdkVersionKey.data(), kTelemetrySdkVersionKey.size())] =
+      Envoy::VersionInfo::version();
+
```

Comment:
This is compliant to the spec in here:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#sdk-provided-resource-attributes

This is part of the default resource which then is merged with other resource coming from detectors (if the user specify any in the config)